api/src/models/team.model.js

```diff
@@ -50,6 +67,18 @@ const teamSchema = new mongoose.Schema(
   { timestamps: true },
 );
 
+// Post middleware/trigger to create associated documents when a team has
+// registered. See: https://stackoverflow.com/a/57971800/15507541.
+// https://mongoosejs.com/docs/middleware.html#post
+teamSchema.post('save', async (doc) => {
+  try {
+    await Website.create({ teamId: doc._id });
+    await Homepage.create({ teamId: doc._id });
```

Comment:
I believe with these additions, these `model` changes are the only ones that we need to make. LMK if it works for you, and feel free to modify this code as I haven't tested thoroughly - it was more of ad hoc research.

Reply:
Could you also cross-check the documents in the db? I just did some minor verification and it appears we have two associated docs creates for each `Website` and `Homepage` models when I tried registering. Pretty weird behavior, but I may have done something wrong.

Reply:
Ahhh okay one of those created may have been from the trigger we made earlier. But we should be able to do without it if we go with this approach.

Reply:
Thanks for that! It works for me. I'll test it properly and go through other controllers as well.

Reply:
I have removed the trigger in MongoDB and other controllers look good to me. Free feel to review this PR again.